docs/source/getting-started/app-architecture.rst

```diff
@@ -82,43 +82,66 @@ The UI
+We take advantage of hooks in particular to share ledger state across components.
+We use custom *DAML React hooks* to query the ledger for contracts, create new contracts, and exercise choices.
 
 .. TODO Link to DAML react hooks API
 
-We can see examples of this in the ``MainView`` component.
-This is the React component that enables the main functionality of the app.
+The ``useState`` hook (not specific to DAML) here keeps track of the user's credentials.
+If they are not set, we render the ``LoginScreen`` with a callback to ``setCredentials``.
+If they are set, then we render the ``MainScreen`` of the app.
+This is wrapped in the ``DamlLedger`` component, a React `context <https://reactjs.org/docs/context.html>`_ with the current view of the ledger.
+
+Let's move on to more advanced uses of our DAML React library.
+The ``MainScreen`` is a simple frame around the ``MainView`` component, which houses the main functionality of our app.
+It uses DAML React hooks to query and update ledger state.
 
 .. literalinclude:: code/ui-before/MainView.tsx
   :language: typescript
   :start-after: // USERS_BEGIN
   :end-before: // USERS_END
 
-For instance, ``allUsers`` uses the query hook to get the ``User`` contracts on the ledger.
-Note however that the query preserves privacy: only users that have added the party currently logged in are shown.
-This is because the observers of a ``User`` contract are exactly the user's friends.
+The ``useParty`` hook simply returns the current user as stored in the ``DamlLedger`` context.
+A more interesting example is the ``allUsers`` line.
+This uses the ``useStreamQuery`` hook to get all ``User`` contracts on the ledger.
+(``User`` is a TypeScript object generated from the DAML code by ``daml codegen``.)
+Note however that this query preserves privacy: only users that have added the current user have their contracts revealed.
+This behaviour is due to the observers on the ``User`` contract being exactly the user's friends.
+
+A final point on this is the *streaming* aspect of the query.
+This means that results are updated as they come in - there is no need for periodic or manual reloading to see updates.
+This functionality comes from a streaming endpoint in the :doc:`HTTP JSON API </json-api/index>`.
 
 .. TODO Explain why you see friends of friends.
 
-Another example is how we exercise the ``AddFriend`` choice of the ``User`` template.
+Another example, showing how to *update* ledger state, is how we exercise the ``AddFriend`` choice of the ``User`` template.
 
 .. literalinclude:: code/ui-before/MainView.tsx
   :language: typescript
   :start-after: // ADDFRIEND_BEGIN
   :end-before: // ADDFRIEND_END
 
-We use the ``useExerciseByKey`` hook to gain access to the ``exerciseAddFriend`` function.
+The ``useExerciseByKey`` hook returns the ``exerciseAddFriend`` function (the ``[]`` around it is simply for ignoring other return values).
```

Comment:
```suggestion
The ``useExerciseByKey`` hook returns the ``exerciseAddFriend`` function.
```
I'll make a PR to remove that stupid boolean return value from the `@daml/react` library immediately.

Reply:
The PR is at https://github.com/digital-asset/daml/pull/4708.